File: backend/src/main/java/edu/gsu/pantherwatch/pantherwatch/api/CourseData.java
package edu.gsu.pantherwatch.pantherwatch.api;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class CourseData {
    @JsonProperty("term")
    private String term;

    @JsonProperty("termDesc")
    private String termDesc;

    @JsonProperty("courseReferenceNumber")
    private String courseReferenceNumber;

    @JsonProperty("partOfTerm")
    private String partOfTerm;

    @JsonProperty("courseNumber")
    private String courseNumber;

    @JsonProperty("subject")
    private String subject;

    @JsonProperty("subjectDescription")
    private String subjectDescription;

    @JsonProperty("sequenceNumber")
    private String sequenceNumber;

    @JsonProperty("campusDescription")
    private String campusDescription;

    @JsonProperty("scheduleTypeDescription")
    private String scheduleTypeDescription;

    @JsonProperty("courseTitle")
    private String courseTitle;

    @JsonProperty("creditHours")
    private int creditHours;

    @JsonProperty("maximumEnrollment")
    private int maximumEnrollment;

    @JsonProperty("enrollment")
    private int enrollment;

    @JsonProperty("seatsAvailable")
    private int seatsAvailable;

    @JsonProperty("waitCapacity")
    private int waitCapacity;

    @JsonProperty("waitCount")
    private int waitCount;

    @JsonProperty("waitAvailable")
    private int waitAvailable;

    @JsonProperty("creditHourHigh")
    private int creditHourHigh;

    @JsonProperty("creditHourLow")
    private int creditHourLow;

    @JsonProperty("linkIdentifier")
    private String linkIdentifier;

    @JsonProperty("isSectionLinked")
    private boolean isSectionLinked;

    @JsonProperty("subjectCourse")
    private String subjectCourse;

    @JsonProperty("faculty")
    private Faculty[] faculty;

    @JsonProperty("meetingsFaculty")
    private MeetingsFaculty[] meetingsFaculty;

    
}

File: frontend-android/app/src/main/java/edu/gsu/pantherwatch/data/model/CourseData.kt
package edu.gsu.pantherwatch.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class CourseData(
    @field:Json(name = "termDesc")
    val termDesc: String?,
    
    @field:Json(name = "courseReferenceNumber")
    val courseReferenceNumber: String?,
    
    @field:Json(name = "partOfTerm")
    val partOfTerm: String?,
    
    @field:Json(name = "courseNumber")
    val courseNumber: String?,
    
    @field:Json(name = "subject")
    val subject: String?,
    
    @field:Json(name = "subjectDescription")
    val subjectDescription: String?,
    
    @field:Json(name = "sequenceNumber")
    val sequenceNumber: String?,
    
    @field:Json(name = "campusDescription")
    val campusDescription: String?,
    
    @field:Json(name = "scheduleTypeDescription")
    val scheduleTypeDescription: String?,
    
    @field:Json(name = "courseTitle")
    val courseTitle: String?,
    
    @field:Json(name = "creditHours")
    val creditHours: Int,
    
    @field:Json(name = "maximumEnrollment")
    val maximumEnrollment: Int,
    
    @field:Json(name = "enrollment")
    val enrollment: Int,
    
    @field:Json(name = "seatsAvailable")
    val seatsAvailable: Int,
    
    @field:Json(name = "waitCapacity")
    val waitCapacity: Int,
    
    @field:Json(name = "waitAvailable")
    val waitAvailable: Int,
    
    @field:Json(name = "creditHourHigh")
    val creditHourHigh: Int,
    
    @field:Json(name = "creditHourLow")
    val creditHourLow: Int,
    
    @field:Json(name = "linkIdentifier")
    val linkIdentifier: String?,
    
    @field:Json(name = "isSectionLinked")
    val isSectionLinked: Boolean,
    
    @field:Json(name = "subjectCourse")
    val subjectCourse: String?,
    
    @field:Json(name = "faculty")
    val faculty: List<Faculty>,
    
    @field:Json(name = "meetingsFaculty")
    val meetingsFaculty: List<MeetingsFaculty>
)
